### PR TITLE
fix: Count unique texts, data leaks in calculate metrics

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist

- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

Added to calculate metadata:
- min length
- max length
- number of unique texts
- test samples in train
Ideas taken from #1049

### CODE REVIEW

File: mteb/abstasks/AbsTask.py [hunk 1]
```diff
@@ -200,7 +200,11 @@ def calculate_metadata_metrics(
 
         descriptive_stats = {}
         hf_subset_stat = "hf_subset_descriptive_stats"
-        pbar_split = tqdm.tqdm(self.metadata.eval_splits, desc="Processing Splits...")
+        eval_splits = self.metadata.eval_splits
+        if self.metadata.type in ["Classification", "MultilabelClassification"]:
+            eval_splits += ["train"]
```
Comment (KennethEnevoldsen) [inline mteb/abstasks/AbsTask.py:205]: Not sure what is going on here?

**Review by KennethEnevoldsen — APPROVED** [review]
All good here - only a minor comment

File: mteb/abstasks/AbsTask.py (hunk 1 above)
Comment (Samoed) [reply mteb/abstasks/AbsTask.py:205]: `Classification` and `MultilabelClassification` have an additional train split, so I think it's important to calculate statistics for it as well

Comment (KennethEnevoldsen) [reply mteb/abstasks/AbsTask.py:205]: Ah yes they are sampled - perfect will merge this in then

**Comment by KennethEnevoldsen** [comment]
Ahh this was merged into main... damn that causes some merge conflicts..

**Comment by Samoed** [comment]
If you are about [2.0](https://github.com/embeddings-benchmark/mteb/pull/1433) I can make PR to update it

**Comment by KennethEnevoldsen** [comment]
That would be great

**Comment by KennethEnevoldsen** [comment]
Merged everything before this PR so that should be solved

**Pull request merged**